src/engagement_db_to_analysis/code_imputation_functions.py

```diff
@@ -0,0 +1,81 @@
+from core_data_modules.util import TimeUtils
+
+from core_data_modules.data_models.code_scheme import CodeTypes
+from core_data_modules.cleaners.cleaning_utils import CleaningUtils
+from core_data_modules.traced_data import Metadata
+from core_data_modules.logging import Logger
+
+
+log = Logger(__name__)
+
+def _impute_age_category(user, messages_traced_data, analysis_dataset_configs):
+    """
+    Imputes age category for age dataset messages.
+
+    :param user: Identifier of user running the pipeline.
+    :type user: str
+    :param messages_traced_data: Messages TracedData objects to impute age_category.
+    :type messages_traced_data: list of TracedData
+    :param analysis_dataset_configs: Analysis dataset configuration in pipeline configuration module.
+    :type analysis_dataset_configs: pipeline_config.analysis_configs.dataset_configurations
+    """
+
+    # Get the coding configurations for age and age_category analysis datasets
+    age_category_cc = None
+    for analysis_dataset_config in analysis_dataset_configs:
+        for coding_config in analysis_dataset_config.coding_configs:
+            if coding_config.age_category_config is not None:
+                age_category_cc = coding_config
+
+    age_cc = None
+    age_engagement_db_datasets = None
+    for analysis_dataset_config in analysis_dataset_configs:
+        for coding_config in analysis_dataset_config.coding_configs:
+            if coding_config.analysis_dataset == age_category_cc.age_category_config.age_analysis_dataset:
+                age_cc = coding_config
+                age_engagement_db_datasets = analysis_dataset_config.engagement_db_datasets
+
+    # Check and impute age_category in age messages only
+    log.info(f"Imputing {age_category_cc.analysis_dataset} for {age_cc.analysis_dataset} messages...")
+    updated_messages_traced_data = []
+    for message in messages_traced_data:
+        if message["dataset"] in age_engagement_db_datasets:
+            age_label = message["labels"][0]
```

Comment:
What if we. have multiple labels assigned to this message? I think you need to get the latest label(s) with this code scheme - there is a function in column_view_conversion that handles this